### PR TITLE
Revise best-practices.md

### PR DESCRIPTION
**Pull request by tasxatzial**

### CODE REVIEW

File: building/tooling/best-practices.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Tooling runs as one-off, short-lived Docker container:
 3. The Docker container is destroyed
```
Comment (BNAndras) [inline building/tooling/best-practices.md]:
```suggestion
3. The Docker container is destroyed.
```

Comment (tasxatzial) [reply building/tooling/best-practices.md]: Hmm, why is a period needed here?

File: building/tooling/best-practices.md [hunk 2]
```diff
@@ -91,7 +91,7 @@ You should try to reduce the image's size, which means that it'll:
 
 - Be faster to deploy
 - Reduce costs for us
-- Improve startup time of each container
+- Improve the startup time of each container
```
Comment (IsaacG) [inline building/tooling/best-practices.md]: The prior line doesn't read "Reduce the cost". I don't think this needs "the" added.

**Review by IsaacG — CHANGES_REQUESTED** [review]

File: building/tooling/best-practices.md (hunk 2 above)
Comment (tasxatzial) [reply building/tooling/best-practices.md]: Sounds reasonable. Removed.

File: building/tooling/best-practices.md (hunk 1 above)
Comment (BNAndras) [reply building/tooling/best-practices.md]: Ah, mistake on my part on mobile. The full context wasn't showing to mark all the steps, but a period should be used after an item in a list if every item is a sentence on its own. See https://apastyle.apa.org/style-grammar-guidelines/lists/numbered and https://www.microsoft.com/en-us/microsoft-365-life-hacks/writing/punctuating-bullet-point-lists.

Comment (BNAndras) [reply building/tooling/best-practices.md]: I can't add a comment to it, but L49 should read "Tooling runs as **a** one-off, short-lived Docker container:"

Comment (BNAndras) [reply building/tooling/best-practices.md]: L61 should read 'Therefore, **it's** fine if the code that runs at build-time is (relatively) slow.'

File: building/tooling/best-practices.md [hunk 3]
```diff
@@ -72,7 +72,7 @@ RUN stack build --resolver lts-20.18 --no-terminal --test --no-run-tests
 ```
 
 First, the `pre-compiled` directory is copied into the image.
-This directory is setup as a sort of fake exercise and depends on the same base libraries that the actual exercise depend on.
+This directory is set up as a sort of fake exercise and depends on the same base libraries that the actual exercise depends on.
```
Comment (BNAndras) [inline building/tooling/best-practices.md]:
```suggestion
This directory is set up as a fake exercise and depends on the same base libraries that the actual exercise depends on.
```

Less confusing this way. It's not a real exercise so it has to be a fake exercise logically. There's no sort of here.

File: building/tooling/best-practices.md [hunk 4]
```diff
@@ -72,7 +72,7 @@ RUN stack build --resolver lts-20.18 --no-terminal --test --no-run-tests
...
 Then we run the tests on that directory, which is similar to how tests are run for an actual exercise.
 Running the tests will result in the base being compiled, but the difference is that this happens at _build time_.
 The resulting Docker image will thus have its base libraries already compiled, which means that no longer has to happen at _run time_, resulting in (much) faster execution times.
```
Comment (BNAndras) [inline building/tooling/best-practices.md]:
```suggestion
The resulting Docker image will thus have its base libraries already compiled, which means that no happens at _run time_, resulting in (much) faster execution times.
```
Awkward phrasing simplified

File: building/tooling/best-practices.md (hunk 1 above)
Comment (BNAndras) [reply building/tooling/best-practices.md]: L120 should read 'The reason "slim" variants are smaller is that they'll have **fewer** features.'

Comment (tasxatzial) [reply building/tooling/best-practices.md]: Good catch. Fixed.

Comment (BNAndras) [reply building/tooling/best-practices.md]: L140 should read 'Distributions that **use** the `apk` package manager (such as Alpine) should use the `--no-cache` flag when using `apk add` to install packages:'

Comment (BNAndras) [reply building/tooling/best-practices.md]: L217 should read 'Its [Dockerfile](https://github.com/exercism/ruby-test-runner/blob/e57ed45b553d6c6411faeea55efa3a4754d1cdbf/Dockerfile) starts with a stage (given the name `build`) that **installs** those packages (via `apk add`) and then installs the **libraries** (via `bundle install`):'

File: building/tooling/best-practices.md (hunk 4 above)
Comment (tasxatzial) [reply building/tooling/best-practices.md]: Looks like a typo on your part. Not sure what you wanted to say, but could we consider changing to "which means that this no longer has to happen" ?

Comment (BNAndras) [reply building/tooling/best-practices.md]:
```suggestion
The resulting Docker image will thus have its base libraries already compiled, which means that no longer happens at _run time_, resulting in (much) faster execution times.
```

My bad, let's try that again.

**Comment by tasxatzial** [comment]
Ready for review

Edit: added the first suggestion (a period should be used after an item in a list if every item is a sentence on its own)

File: building/tooling/best-practices.md [hunk 5]
```diff
@@ -214,7 +214,7 @@ ENTRYPOINT ["/opt/test-runner/bin/run.sh"]
 ##### Example: installing libraries
 
 The Ruby test runner needs the `git`, `openssh`, `build-base`, `gcc` and `wget` packages to be installed before its required libraries (gems) can be installed.
-Its [Dockerfile](https://github.com/exercism/ruby-test-runner/blob/e57ed45b553d6c6411faeea55efa3a4754d1cdbf/Dockerfile) starts with a stage (given the name `build`) that install those packages (via `apk add`) and then installs the libaries (via `bundle install`):
+Its [Dockerfile](https://github.com/exercism/ruby-test-runner/blob/e57ed45b553d6c6411faeea55efa3a4754d1cdbf/Dockerfile) starts with a stage (given the name `build`) that installs those packages (via `apk add`) and then installs the libraries (via `bundle install`):
```
Comment (kotp) [inline building/tooling/best-practices.md]:
```suggestion
Its [Dockerfile](https://github.com/exercism/ruby-test-runner/blob/e57ed45b553d6c6411faeea55efa3a4754d1cdbf/Dockerfile) starts with a stage (given the name `build`) that installs those packages (via `apk add`) and then installs the dependencies (via `bundle install`):
```

Comment (tasxatzial) [reply building/tooling/best-practices.md]: @kotp Fixed!

File: building/tooling/best-practices.md [hunk 6]
```diff
@@ -46,19 +46,19 @@ See the [network docs](/docs/building/tooling/docker#network) for more informati
 
 ### Prefer build-time commands over run-time commands
 
-Tooling runs as one-off, short-lived Docker container:
+Tooling runs as a one-off, short-lived Docker container:
```
Comment (IsaacG) [inline building/tooling/best-practices.md]: If you want to go full-on proper grammar for the whole doc...

```suggestion
The track tooling runs a one-off, short-lived Docker container which executes the following steps.
```

File: building/tooling/best-practices.md [hunk 7]
```diff
@@ -72,10 +72,10 @@ RUN stack build --resolver lts-20.18 --no-terminal --test --no-run-tests
 ```
 
 First, the `pre-compiled` directory is copied into the image.
-This directory is setup as a sort of fake exercise and depends on the same base libraries that the actual exercise depend on.
+This directory is set up as a fake exercise and depends on the same base libraries that the actual exercise depends on.
```
Comment (IsaacG) [inline building/tooling/best-practices.md]:
```suggestion
This directory is set up as a test exercise and depends on the same base libraries that the actual exercise depends on.
```

File: building/tooling/best-practices.md [hunk 8]
```diff
@@ -72,10 +72,10 @@ RUN stack build --resolver lts-20.18 --no-terminal --test --no-run-tests
...
 Then we run the tests on that directory, which is similar to how tests are run for an actual exercise.
 Running the tests will result in the base being compiled, but the difference is that this happens at _build time_.
-The resulting Docker image will thus have its base libraries already compiled, which means that no longer has to happen at _run time_, resulting in (much) faster execution times.
+The resulting Docker image will thus have its base libraries already compiled, which means that no longer happens at _run time_, resulting in (much) faster execution times.
```
Comment (IsaacG) [inline building/tooling/best-practices.md]:
```suggestion
The resulting Docker image will thus have its base libraries already compiled.
This means compiling is not needed at _run time_, resulting in a (much) faster execution.
```

File: building/tooling/best-practices.md [hunk 9]
```diff
@@ -117,7 +117,7 @@ node         20.16.0        1.09GB
 node         20.16.0-slim   219MB
 ```
 
-The reason "slim" variants are smaller is that they'll have less features.
+The reason "slim" variants are smaller is that they'll have fewer features.
```
Comment (IsaacG) [inline building/tooling/best-practices.md]:
```suggestion
The reason "slim" variants are smaller is that they have fewer features.
```

**Review by IsaacG — APPROVED** [review]

File: building/tooling/best-practices.md (hunk 8 above)
Comment (tasxatzial) [reply building/tooling/best-practices.md]: @IsaacG Sure, why not? It's more fun when multiple people are involved.

Looks like an 'at' is missing from this suggestion. Can you confirm?

Comment (IsaacG) [reply building/tooling/best-practices.md]: That is correct. Updated.

File: building/tooling/best-practices.md (hunk 6 above)
Comment (tasxatzial) [reply building/tooling/best-practices.md]: @IsaacG Also, I'm not sure about this one. Is the removal of 'as' intentional?

Comment (IsaacG) [reply building/tooling/best-practices.md]: Yes. The tooling, a collection of code, runs code and executes steps.

**Review by kotp — APPROVED** [review]

**Comment by tasxatzial** [comment]
@BNAndras Whenever you're able, your review is still needed.

**Review by BNAndras — APPROVED** [review]

**Review by SleeplessByte — APPROVED** [review]

**Pull request merged**